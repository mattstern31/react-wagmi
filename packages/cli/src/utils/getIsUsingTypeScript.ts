import { findUp } from 'find-up'

export async function getIsUsingTypeScript() {
  try {
    const cwd = process.cwd()
    const tsconfig = await findUp('tsconfig.json', { cwd })
    if (tsconfig) return true

    const wagmiConfig = await findUp(['wagmi.config.ts', 'wagmi.config.mts'], {
      cwd,
    })
    if (wagmiConfig) return true

    return false
  } catch {
    return false
  }
}
