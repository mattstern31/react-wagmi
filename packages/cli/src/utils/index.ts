export { findConfig } from './findConfig'
export { format } from './format'
export { getAddressDocString } from './getAddressDocString'
export { getIsUsingTypeScript } from './getIsUsingTypeScript'
export { loadEnv } from './loadEnv'
export {
  getIsPackageInstalled,
  getInstallCommand,
  getPackageManager,
} from './packages'
export { resolveConfig } from './resolveConfig'
