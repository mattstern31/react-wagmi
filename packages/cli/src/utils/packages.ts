import { detect } from 'detect-package-manager'
import { execa } from 'execa'

export async function getIsPackageInstalled({
  packageName,
  cwd = process.cwd(),
}: {
  packageName: string
  cwd?: string
}) {
  try {
    const packageManager = await getPackageManager()
    const command =
      packageManager === 'yarn'
        ? ['list', '--pattern', packageName]
        : ['ls', packageName]
    const { stdout } = await execa(packageManager, command, { cwd })
    if (stdout !== '') return true
    return false
  } catch (error) {
    return false
  }
}

export async function getInstallCommand(packageName: string) {
  const packageManager = await getPackageManager()
  switch (packageManager) {
    case 'yarn':
      return [packageManager, ['add', packageName]] as const
    case 'npm':
      return [packageManager, ['install', '--save', packageName]] as const
    case 'pnpm':
      return [packageManager, ['add', packageName]] as const
  }
}

export async function getPackageManager() {
  const userAgent = process.env.npm_config_user_agent
  if (userAgent) {
    if (userAgent.includes('pnpm')) return 'pnpm'
    // The yarn@^3 user agent includes npm, so yarn must be checked first.
    if (userAgent.includes('yarn')) return 'yarn'
    if (userAgent.includes('npm')) return 'npm'
  }
  return detect()
}
