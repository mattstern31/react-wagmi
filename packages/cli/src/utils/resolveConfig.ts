import { bundleRequire } from 'bundle-require'

import type { Config } from '../config'
import type { MaybeArray } from '../types'

type ResolveConfig = {
  /** Path to config file */
  configPath: string
}

/**
 * Bundles and returns wagmi config object from path.
 */
export async function resolveConfig({
  configPath,
}: ResolveConfig): Promise<MaybeArray<Config>> {
  const res = await bundleRequire({
    filepath: configPath,
    ...(process.env.NODE_ENV === 'test'
      ? {
          // FIXME: Workaround https://github.com/egoist/bundle-require/issues/6
          require: (id: string) => import(id),
        }
      : {}),
  })
  let config = res.mod.default
  if (config.default) config = config.default
  if (typeof config !== 'function') return config
  return await config()
}
