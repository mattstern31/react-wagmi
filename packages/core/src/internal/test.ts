export { getWalletClients, testChains } from '../../test'
