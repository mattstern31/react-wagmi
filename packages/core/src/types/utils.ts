/**
 * Converts properties of {@link T} to never
 *
 * @param T - Object to convert
 * @returns Object with properties converted to `never`
 *
 * @example
 * type Result = Never<{ foo: string, bar: number }>
 * //   ^? { foo: never, bar: never }
 */
export type Never<T> = {
  [K in keyof T]: never
}

/**
 * Makes {@link TKeys} optional in {@link TType} while preserving type inference.
 */
// s/o trpc (https://github.com/trpc/trpc/blob/main/packages/server/src/types.ts#L6)
export type PartialBy<TType, TKeys extends keyof TType> = Partial<
  Pick<TType, TKeys>
> &
  Omit<TType, TKeys>
