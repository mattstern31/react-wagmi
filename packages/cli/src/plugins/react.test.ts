import dedent from 'dedent'
import { default as fse } from 'fs-extra'
import { describe, expect, it } from 'vitest'

import { createFixture, typecheck, wagmiAbi } from '../../test'
import { format } from '../utils'
import { react } from './react'

describe('react', () => {
  describe('run', () => {
    it(
      'with TypeScript',
      async () => {
        const { imports, content } = await react().run({
          contracts: [
            {
              name: 'Wagmi',
              abi: wagmiAbi,
              content: '',
              meta: {
                abiName: 'wagmiAbi',
                configName: 'wagmiConfig',
              },
            },
          ],
          isTypeScript: true,
          outputs: [],
        })
        await expect(
          format(`${imports}\n\n${content}`),
        ).resolves.toMatchSnapshot()

        const { paths } = await createFixture({
          copyNodeModules: true,
          files: {
            tsconfig: true,
            'generated.ts': dedent`
              ${imports}

              export const wagmiAbi = ${JSON.stringify(wagmiAbi)} as const

              ${content}
            `,
            'index.ts': dedent`
              import { usePrepareWagmiWrite, useWagmiWrite } from './generated'
    
              const { config } = usePrepareWagmiWrite({
                functionName: 'approve',
                args: ['0xA0Cf798816D4b9b9866b5330EEa46a18382f251e', 123n],
              })
              const { write: preparedWrite } = useWagmiWrite(config)
              preparedWrite?.()
              
              const { write: unpreparedWrite } = useWagmiWrite({
                functionName: 'approve',
              })
              unpreparedWrite({
                args: ['0xA0Cf798816D4b9b9866b5330EEa46a18382f251e', 123n],
              })
            `,
          },
        })

        const tsconfig = await fse.readJSON(paths.tsconfig)
        await fse.writeJSON(paths.tsconfig, {
          ...tsconfig,
          include: [paths['generated.ts'], paths['index.ts']],
        })

        await expect(typecheck(paths.tsconfig)).resolves.toMatchInlineSnapshot(
          '""',
        )
      },
      {
        timeout: 20_000,
      },
    )

    it('without TypeScript', async () => {
      const { imports, content } = await react().run({
        contracts: [
          {
            name: 'Wagmi',
            address: '0xaf0326d92b97df1221759476b072abfd8084f9be',
            abi: wagmiAbi,
            content: '',
            meta: {
              abiName: 'wagmiAbi',
              addressName: 'wagmiAddress',
              configName: 'wagmiConfig',
            },
          },
        ],
        isTypeScript: false,
        outputs: [],
      })
      await expect(
        format(`${imports}\n\n${content}`),
      ).resolves.toMatchSnapshot()
    })
  })
})
