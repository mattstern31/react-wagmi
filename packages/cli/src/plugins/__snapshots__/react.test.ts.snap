// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react > run > with TypeScript 1`] = `
"import {
  useContractRead,
  UseContractReadConfig,
  useContractWrite,
  UseContractWriteConfig,
  usePrepareContractWrite,
  UsePrepareContractWriteConfig,
  useContractEvent,
  UseContractEventConfig,
} from 'wagmi'
import {
  ReadContractResult,
  WriteContractMode,
  PrepareWriteContractResult,
} from 'wagmi/actions'

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function useWagmiRead<
  TFunctionName extends string,
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi'
  > = {} as any,
) {
  return useContractRead({ abi: wagmiAbi, ...config } as UseContractReadConfig<
    typeof wagmiAbi,
    TFunctionName,
    TSelectData
  >)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"balanceOf\\"\`.
 */
export function useWagmiBalanceOf<
  TFunctionName extends 'balanceOf',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'balanceOf',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"getApproved\\"\`.
 */
export function useWagmiGetApproved<
  TFunctionName extends 'getApproved',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'getApproved',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"isApprovedForAll\\"\`.
 */
export function useWagmiIsApprovedForAll<
  TFunctionName extends 'isApprovedForAll',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'isApprovedForAll',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"name\\"\`.
 */
export function useWagmiName<
  TFunctionName extends 'name',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'name',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"ownerOf\\"\`.
 */
export function useWagmiOwnerOf<
  TFunctionName extends 'ownerOf',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'ownerOf',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"supportsInterface\\"\`.
 */
export function useWagmiSupportsInterface<
  TFunctionName extends 'supportsInterface',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'supportsInterface',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"symbol\\"\`.
 */
export function useWagmiSymbol<
  TFunctionName extends 'symbol',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'symbol',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"tokenURI\\"\`.
 */
export function useWagmiTokenUri<
  TFunctionName extends 'tokenURI',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'tokenURI',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"totalSupply\\"\`.
 */
export function useWagmiTotalSupply<
  TFunctionName extends 'totalSupply',
  TSelectData = ReadContractResult<typeof wagmiAbi, TFunctionName>,
>(
  config: Omit<
    UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return useContractRead({
    abi: wagmiAbi,
    functionName: 'totalSupply',
    ...config,
  } as UseContractReadConfig<typeof wagmiAbi, TFunctionName, TSelectData>)
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function useWagmiWrite<
  TFunctionName extends string,
  TMode extends WriteContractMode = undefined,
>(
  config: TMode extends 'prepared'
    ? UseContractWriteConfig<
        PrepareWriteContractResult<typeof wagmiAbi, string>['request']['abi'],
        TFunctionName,
        TMode
      >
    : UseContractWriteConfig<typeof wagmiAbi, TFunctionName, TMode> & {
        abi?: never
      } = {} as any,
) {
  return useContractWrite<typeof wagmiAbi, TFunctionName, TMode>({
    abi: wagmiAbi,
    ...config,
  } as any)
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"approve\\"\`.
 */
export function useWagmiApprove<TMode extends WriteContractMode = undefined>(
  config: TMode extends 'prepared'
    ? UseContractWriteConfig<
        PrepareWriteContractResult<
          typeof wagmiAbi,
          'approve'
        >['request']['abi'],
        'approve',
        TMode
      > & { functionName?: 'approve' }
    : UseContractWriteConfig<typeof wagmiAbi, 'approve', TMode> & {
        abi?: never
        functionName?: 'approve'
      } = {} as any,
) {
  return useContractWrite<typeof wagmiAbi, 'approve', TMode>({
    abi: wagmiAbi,
    functionName: 'approve',
    ...config,
  } as any)
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"mint\\"\`.
 */
export function useWagmiMint<TMode extends WriteContractMode = undefined>(
  config: TMode extends 'prepared'
    ? UseContractWriteConfig<
        PrepareWriteContractResult<typeof wagmiAbi, 'mint'>['request']['abi'],
        'mint',
        TMode
      > & { functionName?: 'mint' }
    : UseContractWriteConfig<typeof wagmiAbi, 'mint', TMode> & {
        abi?: never
        functionName?: 'mint'
      } = {} as any,
) {
  return useContractWrite<typeof wagmiAbi, 'mint', TMode>({
    abi: wagmiAbi,
    functionName: 'mint',
    ...config,
  } as any)
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"safeTransferFrom\\"\`.
 */
export function useWagmiSafeTransferFrom<
  TMode extends WriteContractMode = undefined,
>(
  config: TMode extends 'prepared'
    ? UseContractWriteConfig<
        PrepareWriteContractResult<
          typeof wagmiAbi,
          'safeTransferFrom'
        >['request']['abi'],
        'safeTransferFrom',
        TMode
      > & { functionName?: 'safeTransferFrom' }
    : UseContractWriteConfig<typeof wagmiAbi, 'safeTransferFrom', TMode> & {
        abi?: never
        functionName?: 'safeTransferFrom'
      } = {} as any,
) {
  return useContractWrite<typeof wagmiAbi, 'safeTransferFrom', TMode>({
    abi: wagmiAbi,
    functionName: 'safeTransferFrom',
    ...config,
  } as any)
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"setApprovalForAll\\"\`.
 */
export function useWagmiSetApprovalForAll<
  TMode extends WriteContractMode = undefined,
>(
  config: TMode extends 'prepared'
    ? UseContractWriteConfig<
        PrepareWriteContractResult<
          typeof wagmiAbi,
          'setApprovalForAll'
        >['request']['abi'],
        'setApprovalForAll',
        TMode
      > & { functionName?: 'setApprovalForAll' }
    : UseContractWriteConfig<typeof wagmiAbi, 'setApprovalForAll', TMode> & {
        abi?: never
        functionName?: 'setApprovalForAll'
      } = {} as any,
) {
  return useContractWrite<typeof wagmiAbi, 'setApprovalForAll', TMode>({
    abi: wagmiAbi,
    functionName: 'setApprovalForAll',
    ...config,
  } as any)
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"transferFrom\\"\`.
 */
export function useWagmiTransferFrom<
  TMode extends WriteContractMode = undefined,
>(
  config: TMode extends 'prepared'
    ? UseContractWriteConfig<
        PrepareWriteContractResult<
          typeof wagmiAbi,
          'transferFrom'
        >['request']['abi'],
        'transferFrom',
        TMode
      > & { functionName?: 'transferFrom' }
    : UseContractWriteConfig<typeof wagmiAbi, 'transferFrom', TMode> & {
        abi?: never
        functionName?: 'transferFrom'
      } = {} as any,
) {
  return useContractWrite<typeof wagmiAbi, 'transferFrom', TMode>({
    abi: wagmiAbi,
    functionName: 'transferFrom',
    ...config,
  } as any)
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function usePrepareWagmiWrite<TFunctionName extends string>(
  config: Omit<
    UsePrepareContractWriteConfig<typeof wagmiAbi, TFunctionName>,
    'abi'
  > = {} as any,
) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    ...config,
  } as UsePrepareContractWriteConfig<typeof wagmiAbi, TFunctionName>)
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"approve\\"\`.
 */
export function usePrepareWagmiApprove(
  config: Omit<
    UsePrepareContractWriteConfig<typeof wagmiAbi, 'approve'>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    functionName: 'approve',
    ...config,
  } as UsePrepareContractWriteConfig<typeof wagmiAbi, 'approve'>)
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"mint\\"\`.
 */
export function usePrepareWagmiMint(
  config: Omit<
    UsePrepareContractWriteConfig<typeof wagmiAbi, 'mint'>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    functionName: 'mint',
    ...config,
  } as UsePrepareContractWriteConfig<typeof wagmiAbi, 'mint'>)
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"safeTransferFrom\\"\`.
 */
export function usePrepareWagmiSafeTransferFrom(
  config: Omit<
    UsePrepareContractWriteConfig<typeof wagmiAbi, 'safeTransferFrom'>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    functionName: 'safeTransferFrom',
    ...config,
  } as UsePrepareContractWriteConfig<typeof wagmiAbi, 'safeTransferFrom'>)
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"setApprovalForAll\\"\`.
 */
export function usePrepareWagmiSetApprovalForAll(
  config: Omit<
    UsePrepareContractWriteConfig<typeof wagmiAbi, 'setApprovalForAll'>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    functionName: 'setApprovalForAll',
    ...config,
  } as UsePrepareContractWriteConfig<typeof wagmiAbi, 'setApprovalForAll'>)
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"transferFrom\\"\`.
 */
export function usePrepareWagmiTransferFrom(
  config: Omit<
    UsePrepareContractWriteConfig<typeof wagmiAbi, 'transferFrom'>,
    'abi' | 'functionName'
  > = {} as any,
) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    functionName: 'transferFrom',
    ...config,
  } as UsePrepareContractWriteConfig<typeof wagmiAbi, 'transferFrom'>)
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function useWagmiEvent<TEventName extends string>(
  config: Omit<
    UseContractEventConfig<typeof wagmiAbi, TEventName>,
    'abi'
  > = {} as any,
) {
  return useContractEvent({
    abi: wagmiAbi,
    ...config,
  } as UseContractEventConfig<typeof wagmiAbi, TEventName>)
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__ and \`eventName\` set to \`\\"Approval\\"\`.
 */
export function useWagmiApprovalEvent(
  config: Omit<
    UseContractEventConfig<typeof wagmiAbi, 'Approval'>,
    'abi' | 'eventName'
  > = {} as any,
) {
  return useContractEvent({
    abi: wagmiAbi,
    eventName: 'Approval',
    ...config,
  } as UseContractEventConfig<typeof wagmiAbi, 'Approval'>)
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__ and \`eventName\` set to \`\\"ApprovalForAll\\"\`.
 */
export function useWagmiApprovalForAllEvent(
  config: Omit<
    UseContractEventConfig<typeof wagmiAbi, 'ApprovalForAll'>,
    'abi' | 'eventName'
  > = {} as any,
) {
  return useContractEvent({
    abi: wagmiAbi,
    eventName: 'ApprovalForAll',
    ...config,
  } as UseContractEventConfig<typeof wagmiAbi, 'ApprovalForAll'>)
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__ and \`eventName\` set to \`\\"Transfer\\"\`.
 */
export function useWagmiTransferEvent(
  config: Omit<
    UseContractEventConfig<typeof wagmiAbi, 'Transfer'>,
    'abi' | 'eventName'
  > = {} as any,
) {
  return useContractEvent({
    abi: wagmiAbi,
    eventName: 'Transfer',
    ...config,
  } as UseContractEventConfig<typeof wagmiAbi, 'Transfer'>)
}
"
`;

exports[`react > run > without TypeScript 1`] = `
"import {
  useContractRead,
  useContractWrite,
  usePrepareContractWrite,
  useContractEvent,
} from 'wagmi'

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function useWagmiRead(config = {}) {
  return useContractRead({ abi: wagmiAbi, address: wagmiAddress, ...config })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"balanceOf\\"\`.
 */
export function useWagmiBalanceOf(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'balanceOf',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"getApproved\\"\`.
 */
export function useWagmiGetApproved(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'getApproved',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"isApprovedForAll\\"\`.
 */
export function useWagmiIsApprovedForAll(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'isApprovedForAll',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"name\\"\`.
 */
export function useWagmiName(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'name',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"ownerOf\\"\`.
 */
export function useWagmiOwnerOf(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'ownerOf',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"supportsInterface\\"\`.
 */
export function useWagmiSupportsInterface(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'supportsInterface',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"symbol\\"\`.
 */
export function useWagmiSymbol(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'symbol',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"tokenURI\\"\`.
 */
export function useWagmiTokenUri(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'tokenURI',
    ...config,
  })
}

/**
 * Wraps __{@link useContractRead}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"totalSupply\\"\`.
 */
export function useWagmiTotalSupply(config = {}) {
  return useContractRead({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'totalSupply',
    ...config,
  })
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function useWagmiWrite(config = {}) {
  return useContractWrite({ abi: wagmiAbi, address: wagmiAddress, ...config })
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"approve\\"\`.
 */
export function useWagmiApprove(config = {}) {
  return useContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'approve',
    ...config,
  })
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"mint\\"\`.
 */
export function useWagmiMint(config = {}) {
  return useContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'mint',
    ...config,
  })
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"safeTransferFrom\\"\`.
 */
export function useWagmiSafeTransferFrom(config = {}) {
  return useContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'safeTransferFrom',
    ...config,
  })
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"setApprovalForAll\\"\`.
 */
export function useWagmiSetApprovalForAll(config = {}) {
  return useContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'setApprovalForAll',
    ...config,
  })
}

/**
 * Wraps __{@link useContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"transferFrom\\"\`.
 */
export function useWagmiTransferFrom(config = {}) {
  return useContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'transferFrom',
    ...config,
  })
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function usePrepareWagmiWrite(config = {}) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    ...config,
  })
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"approve\\"\`.
 */
export function usePrepareWagmiApprove(config = {}) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'approve',
    ...config,
  })
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"mint\\"\`.
 */
export function usePrepareWagmiMint(config = {}) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'mint',
    ...config,
  })
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"safeTransferFrom\\"\`.
 */
export function usePrepareWagmiSafeTransferFrom(config = {}) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'safeTransferFrom',
    ...config,
  })
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"setApprovalForAll\\"\`.
 */
export function usePrepareWagmiSetApprovalForAll(config = {}) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'setApprovalForAll',
    ...config,
  })
}

/**
 * Wraps __{@link usePrepareContractWrite}__ with \`abi\` set to __{@link wagmiAbi}__ and \`functionName\` set to \`\\"transferFrom\\"\`.
 */
export function usePrepareWagmiTransferFrom(config = {}) {
  return usePrepareContractWrite({
    abi: wagmiAbi,
    address: wagmiAddress,
    functionName: 'transferFrom',
    ...config,
  })
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__.
 */
export function useWagmiEvent(config = {}) {
  return useContractEvent({ abi: wagmiAbi, address: wagmiAddress, ...config })
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__ and \`eventName\` set to \`\\"Approval\\"\`.
 */
export function useWagmiApprovalEvent(config = {}) {
  return useContractEvent({
    abi: wagmiAbi,
    address: wagmiAddress,
    eventName: 'Approval',
    ...config,
  })
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__ and \`eventName\` set to \`\\"ApprovalForAll\\"\`.
 */
export function useWagmiApprovalForAllEvent(config = {}) {
  return useContractEvent({
    abi: wagmiAbi,
    address: wagmiAddress,
    eventName: 'ApprovalForAll',
    ...config,
  })
}

/**
 * Wraps __{@link useContractEvent}__ with \`abi\` set to __{@link wagmiAbi}__ and \`eventName\` set to \`\\"Transfer\\"\`.
 */
export function useWagmiTransferEvent(config = {}) {
  return useContractEvent({
    abi: wagmiAbi,
    address: wagmiAddress,
    eventName: 'Transfer',
    ...config,
  })
}
"
`;
